lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -35,58 +35,40 @@ def serializable_hash(options = {})
         private
 
         def add_links(resource, name, serializers)
-          type = serialized_object_type(serializers)
           resource[:links] ||= {}
-
-          if name.to_s == type || !type
-            resource[:links][name] ||= []
-            resource[:links][name] += serializers.map{|serializer| serializer.id.to_s }
-          else
-            resource[:links][name] ||= {}
-            resource[:links][name][:type] = type
-            resource[:links][name][:ids] ||= []
-            resource[:links][name][:ids] += serializers.map{|serializer| serializer.id.to_s }
-          end
+          resource[:links][name] ||= { linkage: [] }
+          resource[:links][name][:linkage] += serializers.map { |serializer| { type: serializer.type, id: serializer.id.to_s } }
         end
 
         def add_link(resource, name, serializer)
           resource[:links] ||= {}
-          resource[:links][name] = nil
+          resource[:links][name] = { linkage: nil }
```

Comment:
What if resource has no links. Will it still have `{ linkage: nil }`? I believe that we should not have links if we have no associations. What do you think?